File: Quivi.Backoffice.Api/Dtos/ModifierGroup.cs
using Quivi.Domain.Entities;

namespace Quivi.Backoffice.Api.Dtos
{
    public class ModifierGroup
    {
        public required string Id { get; init; }
        public required string Name { get; init; }
        public int MinSelection { get; init; }
        public int MaxSelection { get; init; }
        public required IReadOnlyDictionary<string, ModifierGroupItem> Items { get; init; }
        public required IReadOnlyDictionary<Language, ModifierGroupTranslation> Translations { get; init; }
    }

    public class ModifierGroupItem
    {
        public decimal Price { get; init; }
        public int SortIndex { get; init; }
    }

    public class ModifierGroupTranslation
    {
        public required string Name { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/modifierGroups/ModifierGroup.ts
import { Language } from "../Language";

export interface ModifierGroup {
    readonly id: string;
    readonly name: string;
    readonly minSelection: number;
    readonly maxSelection: number;
    readonly items: Record<string, ModifierGroupItem>;
    readonly translations: Record<Language, ModifierGroupTranslation>;
}

export interface ModifierGroupItem {
    readonly price: number;
    readonly sortIndex: number;
}

export interface ModifierGroupTranslation {
    readonly name: string;
}